src/bin/pg_autoctl/pgsql.c

```diff
@@ -2421,46 +2444,46 @@ pgsql_get_postgres_metadata(PGSQL *pgsql,
-		 * and report that.
-		 */
-		"select pg_is_in_recovery(),"
-		" coalesce(rep.sync_state, '') as sync_state,"
-		" case when pg_is_in_recovery()"
-		" then coalesce(pg_last_wal_receive_lsn(), pg_last_wal_replay_lsn())"
-		" else pg_current_wal_flush_lsn()"
-		" end as current_lsn,"
-		" pg_control_version, catalog_version_no, system_identifier"
-		" from (values(1)) as dummy"
-		" full outer join"
-		" (select pg_control_version, catalog_version_no, system_identifier "
-		"    from pg_control_system()"
-		" )"
-		" as control on true"
-		" full outer join"
-		" ("
-		"   select sync_state"
-		"     from pg_replication_slots slot"
-		"     join pg_stat_replication rep"
-		"       on rep.pid = slot.active_pid"
-		"   where slot_name ~ '" REPLICATION_SLOT_NAME_PATTERN "' "
-		"      or slot_name = '" REPLICATION_SLOT_NAME_DEFAULT "' "
-		"order by case sync_state "
-		"         when 'quorum' then 4 "
-		"         when 'sync' then 3 "
-		"         when 'potential' then 2 "
-		"         when 'async' then 1 "
-		"         else 0 end "
-		"    desc limit 1"
-		" ) "
-		"as rep on true";
+    char *sql =
+            /*
+             * Make it so that we still have the current WAL LSN even in the case
+             * where there's no replication slot in use by any standby.
+             *
+             * When on the primary, we might have multiple standby nodes connected.
+             * We're good when at least one of them is either 'sync' or 'quorum'.
+             * We don't check individual replication slots, we take the "best" one
+             * and report that.
+             */
+            "select pg_is_in_recovery(),"
+            " coalesce(rep.sync_state, '') as sync_state,"
+            " case when pg_is_in_recovery()"
+            " then coalesce(pg_last_wal_receive_lsn(), pg_last_wal_replay_lsn())"
+            " else pg_current_wal_flush_lsn()"
+            " end as current_lsn,"
+            " pg_control_version, catalog_version_no, system_identifier"
+            " from (values(1)) as dummy"
+            " full outer join"
+            " (select pg_control_version, catalog_version_no, system_identifier "
+            "    from pg_control_system()"
+            " )"
+            " as control on true"
+            " full outer join"
+            " ("
+            "   select sync_state"
+            "     from pg_replication_slots slot"
+            "     join pg_stat_replication rep"
+            "       on rep.pid = slot.active_pid"
+            "   where slot_name ~ '" REPLICATION_SLOT_NAME_PATTERN "' "
+            "      or slot_name = '" REPLICATION_SLOT_NAME_DEFAULT "' "
+            "order by case sync_state "
+            "         when 'quorum' then 4 "
+            "         when 'sync' then 3 "
+            "         when 'potential' then 2 "
+            "         when 'async' then 1 "
+            "         else 0 end "
+            "    desc limit 1"
+            " ) "
+            "as rep on true";
```

Comment:
This looks like a whitespace change that we should not have in this PR, please fix.